source/constructs/api/discovery_job/crud.py

```diff
@@ -104,15 +103,28 @@ def get_running_run(job_id: int) -> models.DiscoveryJobRun:
     return db_run
 
 
+def __get_database_name_key(database_type: str) -> str:
+    if database_type == DatabaseType.S3.value:
+        return "bucket_name"
+    elif database_type == DatabaseType.RDS.value:
+        return "instance_id"
+    elif database_type == DatabaseType.GLUE.value:
+        return "glue_database_name"
+    else:
+        return "instance_id"
+
+
 def __add_job_databases(run: models.DiscoveryJobRun, database_type: str, base_time_dict: dict):
-    databases = get_catalog_database_level_classification_by_type_all(database_type).all()
-    for database in databases:
-        base_time = base_time_dict.get(f'{database.account_id}-{database.region}-{database_type}-{database.database_name}')
+    data_sources = list_resources_by_database_type(database_type).all()
+    database_name_key = __get_database_name_key(database_type)
+    for data_source in data_sources:
+        database_name = getattr(data_source, database_name_key)
+        base_time = base_time_dict.get(f'{data_source.account_id}-{data_source.region}-{database_type}-{database_name}')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You are using the `get` method without a default argument to return the value of a key in a dictionary. We recommended that you use a default argument so that if the value for your key is not found, a default value is returned. If a default value is not provided and the key is not found, then `None` is returned.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)